pkgs/by-name/cw/cwtch/package.nix

```diff
@@ -0,0 +1,54 @@
+{
+  buildGoModule,
+  fetchgit,
+  lib,
+}:
+buildGoModule rec {
+  pname = "libcwtch";
+  version = "0.1.3";
+  src = fetchgit {
+    url = "https://git.openprivacy.ca/cwtch.im/autobindings.git";
+    rev = "v${version}";
+    hash = "sha256-Gp6JnyjKChB7w0fUHkreu81QRXQ3YdMb2ReD/VCkVOE=";
+  };
+
+  vendorHash = "sha256-UgG/yjupUFrfjPZ4E+OM1VsWi9MuMuHxcZ4yvz+q/Y0=";
+  overrideModAttrs = (
+    old: {
+      preBuild = ''
+        make lib.go
+      '';
+    }
+  );
+
+  postPatch = ''
+    substituteInPlace Makefile \
+      --replace-fail '$(shell git describe --tags)' v${version} \
+      --replace-fail '$(shell git log -1 --format=%cd --date=format:%G-%m-%d-%H-%M)' 1980-01-01-00-00
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+    make linux
+    runHook postBuild
+  '';
```

Comment:
```suggestion
  makeFlags = [ "linux" ];
```

Reply:
I tried this and it doesn't work because it doesn't run any of the custom build infrastructure in this package.

Reply:
It should be equivalent. What error did you get?

Reply:
```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/j6h6qr5n33avwx6mjfrz5c732165ikdz-autobindings
source root is autobindings
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Building subPackage ./constants
Building subPackage ./experiments/server_hosting
Building subPackage ./features
Building subPackage ./generate
Building subPackage ./templates
templates/lib_template.go:36:2: missing import path
```

Reply:
Yes, `buildGoModule` doesn't support disabling go build phase.